pkg/client/interface.go

```diff
@@ -375,8 +375,8 @@ type QueryCache[T any] interface {
 // at multiple heights for a given key.
 type HistoricalQueryCache[T any] interface {
 	QueryCache[T]
-	// GetAtHeight retrieves the nearest value <= the specified height
-	GetAtHeight(key string, height int64) (T, error)
-	// SetAtHeight adds or updates a value at a specific height
-	SetAtHeight(key string, value T, height int64) error
+	// GetAsOfVersion retrieves the nearest value <= the specified version number.
+	GetAsOfVersion(key string, version int64) (T, error)
```

Comment:
Can we go with one of the following instead:

1. GetVersioned
2. GetAtVersion
3. GetHistorical

Don't care which one but `AsOf` in a function name just doesn't feel right.

Reply:
In that case, `GetVersion` would be my preference. Any objections @red-0ne @Olshansk?

Reply:
Also adding `GetLatestVersion`.